### PR TITLE
Created dialog component

### PR DESCRIPTION
**Pull request by josephchimebuka**
I created a screen called  Dialog in the screen directory to view my created dialog components.
 To see the various dialog boxes, 
- click on the notification bell at the bottom of the screen which will show a button that says open dialog page then you will see various buttons for the different dialog boxes each

### CODE REVIEW

**Comment by ugur-eren** [comment]
Hi! It would be better to get the buttons from the props as described in the issue, instead of defining button1 and button2 hardcoded.

Dialog component should have a single prop `buttons`. `{label: string; type: dangerous | primary | secondary; onPress: () => void}[]`

**Comment by josephchimebuka** [comment]
I have added a single button prop to the code

File: JoyboyCommunity/assets/svgs/svgComponents/Exit.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import * as React from "react"
+import Svg, { SvgProps, Rect, Path } from "react-native-svg"
+const Exit = (props: SvgProps) => (
```
Comment (ugur-eren) [inline JoyboyCommunity/assets/svgs/svgComponents/Exit.tsx:3]: SVGs should be moved to [icons.tsx](https://github.com/keep-starknet-strange/joyboy/blob/5264c5680e32db71bf8b18f5d491d972925ab64f/JoyboyCommunity/src/assets/icons.tsx) file.

File: JoyboyCommunity/assets/svgs/svgComponents/Exit.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    height={56}
+    fill="none"
+    {...props}
+  >
+    <Rect
+      width={56}
+      height={56}
+      x={0.5}
+      fill="#EC796B"
+      fillOpacity={0.1}
```
Comment (ugur-eren) [inline JoyboyCommunity/assets/svgs/svgComponents/Exit.tsx:15]: colors should be currentColor, and the usage should include the color, not the definition. For example, this should be `fill="currentColor"` and usage would be `<ExitIcon color="#EC796B" />`

File: JoyboyCommunity/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,15954 @@
+{
```
Comment (ugur-eren) [inline JoyboyCommunity/package-lock.json:1]: We're using yarn, not npm. package-lock file should be removed.

File: JoyboyCommunity/src/components/modal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  ModalContainer,
+  Overlay,
+  StyledButton,
+  Title,
+} from './styled';
+
+interface Button {
+  label: string;
+  type: 'dangerous' | 'primary' | 'secondary';
+  onPress: () => void;
```
Comment (ugur-eren) [inline JoyboyCommunity/src/components/modal/index.tsx:19]: No need to create an interface for button

File: JoyboyCommunity/src/components/modal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+}
+interface CustomModalProps {
+  name: string;
+  buttons: Button[];
+  icon: React.ReactNode;
+  description: string;
+  visible: boolean;
+  // onClose: () => void;
+  color?: string;
```
Comment (ugur-eren) [inline JoyboyCommunity/src/components/modal/index.tsx:28]: seems like `color` prop is not used anymore, can be removed

File: JoyboyCommunity/src/components/modal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            <Line />
+            <IconContainer>{icon}</IconContainer>
+            <Title>{name}</Title>
+            <Content>{description}</Content>
+          </Container>
+          <Container>
+            {buttons.map((button, index) => (
+              <StyledButton key={index} type={button.type} onPress={button.onPress}>
+                <ButtonText type={button.type}>{button.label}</ButtonText>
+              </StyledButton>
```
Comment (ugur-eren) [inline JoyboyCommunity/src/components/modal/index.tsx:46]: We already have a component for buttons, [`Button`](https://github.com/keep-starknet-strange/joyboy/blob/5264c5680e32db71bf8b18f5d491d972925ab64f/JoyboyCommunity/src/components/button/index.tsx) should be used instead of using a custom one.

File: JoyboyCommunity/src/modules/post/styled.ts
Comment (ugur-eren) [inline JoyboyCommunity/src/modules/post/styled.ts:1]: Why have an empty file?

**Review by ugur-eren — CHANGES_REQUESTED** [review]
Thanks for the work, just requested some minor changes.

**Comment by josephchimebuka** [comment]
I have adjusted the changes with the ones you reviewed

File: JoyboyCommunity/src/components/modal/index.tsx (hunk 4 above)
Comment (josephchimebuka) [reply JoyboyCommunity/src/components/modal/index.tsx:19]: The component is a Touchableopacity and not a button

File: JoyboyCommunity/src/assets/icons.tsx [hunk 7]
```diff
@@ -107,6 +107,228 @@ export const UserIcon: React.FC<SvgProps> = (props) => (
   </Svg>
 );
 
+export const CopyIcon: React.FC<SvgProps> = (props) => (
```
Comment (ugur-eren) [inline JoyboyCommunity/src/assets/icons.tsx:110]: CopyIcon has duplicate, there is already a CopyIcon in this file

File: JoyboyCommunity/src/assets/icons.tsx [hunk 8]
```diff
@@ -107,6 +107,228 @@ export const UserIcon: React.FC<SvgProps> = (props) => (
...
+    />
+    <Path
+      fill="#EC796B"
+      fillRule="evenodd"
+      d="M26.862 18.015a5.195 5.195 0 0 1 3.276 0c1.06.352 1.892 1.186 2.729 2.359.833 1.169 1.754 2.798 2.94 4.899l.047.082c1.188 2.1 2.108 3.73 2.682 5.052.577 1.327.864 2.474.635 3.575a5.408 5.408 0 0 1-1.628 2.88c-.827.76-1.952 1.08-3.37 1.235-1.409.153-3.252.153-5.624.153h-.098c-2.372 0-4.216 0-5.625-.153-1.417-.154-2.542-.475-3.369-1.235a5.407 5.407 0 0 1-1.628-2.88c-.23-1.101.058-2.248.635-3.575.574-1.322 1.494-2.951 2.682-5.052l.046-.082c1.187-2.1 2.108-3.73 2.942-4.9.836-1.172 1.668-2.006 2.728-2.358ZM27.5 33c0-.552.446-1 .995-1h.01c.55 0 .995.448.995 1s-.446 1-.995 1h-.01a.998.998 0 0 1-.995-1Zm0-4a1 1 0 1 0 2 0v-4a1 1 0 1 0-2 0v4Z"
+      clipRule="evenodd"
+    />
+  </Svg>
+)
+=======
```
Comment (ugur-eren) [inline JoyboyCommunity/src/assets/icons.tsx:331]: There is a git conflict marker left

File: JoyboyCommunity/src/components/modal/index.tsx (hunk 4 above)
Comment (ugur-eren) [reply JoyboyCommunity/src/components/modal/index.tsx:19]: I'm addressing the `interface Button` it should be removed and moved into the props directly.

```ts
export type CustomModalProps = {
  name: string;
  buttons: {
    label: string;
    type: 'dangerous' | 'primary' | 'secondary';
    onPress: () => void;
  }[];
  icon: React.ReactNode;
  description: string;
  visible: boolean;
}
```

File: JoyboyCommunity/src/components/modal/index.tsx (hunk 6 above)
Comment (ugur-eren) [reply JoyboyCommunity/src/components/modal/index.tsx:44]: As I've stated before, please use the Button component instead of using custom one.

File: JoyboyCommunity/src/components/modal/styled.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  width: 281px;
+  height: 54px;
+  border-radius: 40px;
+`;
+
+
+
+
+
+const getTextColor = (type: string) => {
```
Comment (ugur-eren) [inline JoyboyCommunity/src/components/modal/styled.ts:81]: Please fix the formatting

File: JoyboyCommunity/src/screens/Auth/Login.tsx [hunk 10]
```diff
@@ -14,3 +14,4 @@ export const Login: React.FC = () => {
     </Auth>
   );
 };
+
```
Comment (ugur-eren) [inline JoyboyCommunity/src/screens/Auth/Login.tsx:17]: Please fix the formatting

File: JoyboyCommunity/src/screens/CreatePost/index.tsx [hunk 11]
```diff
@@ -2,10 +2,8 @@ import {useNavigation} from '@react-navigation/native';
 import {useState} from 'react';
 import {Pressable, TouchableOpacity, View} from 'react-native';
 import {TextInput} from 'react-native-gesture-handler';
-
-import {CopyIcon, GalleryIcon, GifIcon, SendIcon, UserIcon} from '../../assets/icons';
 import {Divider, KeyboardAvoidingView, Typography} from '../../components';
-import {useSendNote} from '../../hooks';
+import {useSendNote} from '../../hooks/useNostr';
```
Comment (ugur-eren) [inline JoyboyCommunity/src/screens/CreatePost/index.tsx:6]: useNostr hook is removed, please update your branch

File: JoyboyCommunity/src/screens/CreatePost/index.tsx
Comment (ugur-eren) [inline JoyboyCommunity/src/screens/CreatePost/index.tsx:1]: Why change unrelated file?

File: JoyboyCommunity/src/assets/icons.tsx [hunk 12]
```diff
@@ -107,6 +107,228 @@ export const UserIcon: React.FC<SvgProps> = (props) => (
...
+      strokeLinecap="round"
+      strokeLinejoin="round"
+      strokeWidth={1.5}
+      d="M2.75 12c0-1.84 1.506-3.333 3.364-3.333.672 0 1.464.116 2.117-.057a1.674 1.674 0 0 0 1.19-1.179c.174-.647.057-1.432.057-2.098C9.478 3.493 10.984 2 12.84 2M12.842 18h2.523m-4.949-4.15c-.126-.8-.281-.801-1.61-.85h-1.01c-.557 0-1.009.448-1.009 1v3c0 .552.452 1 1.01 1h1.816c.39 0 .803-.313.803-.7v-1.1c0-.11-.113-.304-.224-.304H9.068M12.842 13h1.261m0 0h1.262m-1.262 0v4.875M21.251 13h-2.523c-.557 0-1.009.448-1.009 1v1.5m0 0V18m0-2.5h2.523"
+    />
+  </Svg>
+);
+
+
+export const Line: React.FC<SvgProps> = (props) => (
```
Comment (ugur-eren) [inline JoyboyCommunity/src/assets/icons.tsx:220]: Icon names should end in Icon. `LineIcon` instead of `Line`

File: JoyboyCommunity/src/screens/Notifications/Notification/index.tsx
Comment (ugur-eren) [inline JoyboyCommunity/src/screens/Notifications/Notification/index.tsx:1]: I don't understand the changes in this file

File: JoyboyCommunity/src/types/routes.ts
Comment (ugur-eren) [inline JoyboyCommunity/src/types/routes.ts:1]: Please update your branch. These changes are related to old version of this file.

File: JoyboyCommunity/assets/Logo.png
Comment (ugur-eren) [inline JoyboyCommunity/assets/Logo.png:1]: Where does this image used?

**Review by ugur-eren — CHANGES_REQUESTED** [review]
Hi, I've requested some changes. Please take a look at each one. And please make sure the app runs without errors before requesting a new review

**Comment by ugur-eren** [comment]
Closing in favour of #144
